src/spatial/detail/ArborX_CrsGraphWrapperImpl.hpp

```diff
@@ -109,6 +109,33 @@ struct InsertGenerator
 namespace CrsGraphWrapperImpl
 {
 
+#ifdef KOKKOS_COMPILER_NVCC
+// FIXME_NVCC: Workaround for a segfault observed with some NVIDIA compilers
+// when using parallel_for directly. See #1297.
+template <typename S, typename T>
+class NVCCSegfaultWorkaround
+{
+public:
+  template <typename ExecutionSpace>
+  NVCCSegfaultWorkaround(ExecutionSpace const &space, S s, T t)
+  {
+    static_assert(Kokkos::is_execution_space_v<ExecutionSpace>);
+    static_assert(Kokkos::is_view_v<S>);
+    static_assert(Kokkos::is_view_v<T>);
+    Kokkos::parallel_for("ArborX::CrsGraphWrapper::copy_" + s.label() + "_" +
+                             t.label(),
+                         Kokkos::RangePolicy(space, 0, s.extent(0)), *this);
+  }
+
+  KOKKOS_FUNCTION
+  void operator()(int i) const { _s(i) = _t(i); }
+
+private:
+  S _s;
+  T _t;
```

Comment:
I'd prefer if you got rid of all public and private access specifiers and moved these data member definition to the top (for readability)